src/character-classes/compiler.ts

```diff
@@ -0,0 +1,24 @@
+import type { CharacterClass } from '../types';
+
+export function compileCharacterClass({ characters }: CharacterClass): string {
+  if (characters.length === 0) {
+    return '';
+  }
+
+  if (characters.length === 1) {
+    return characters[0]!;
+  }
+
+  return `[${escapeHypen(characters).join('')}]`;
+}
+
+// If passed characters includes hypen (`-`) it need to be moved to
+// first (or last) place in order to treat it as hypen character and not a range.
+// See: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_expressions/Character_classes#types
+function escapeHypen(characters: string[]) {
+  if (characters.includes('-')) {
+    return ['-', ...characters.filter((c) => c !== '-')];
```

Comment:
Shouldn't this function also call `.join()` and return a string?

Reply:
Not yet, it is parsing it's input for RegexElement. Join will be done when we will compile it